packages/pouchdb-adapter-http/src/index.js

```diff
@@ -71,9 +71,26 @@ function preprocessAttachments(doc) {
   }));
 }
 
+function hasUrlPrefix(opts) {
+  if (!opts.prefix) {
+    return false;
+  }
+
+  var protocol = parseUri(opts.prefix).protocol;
+
+  return protocol === 'http' || protocol === 'https';
+}
+
 // Get all the information you possibly can about the URI given by name and
 // return it as a suitable object.
-function getHost(name) {
+function getHost(name, opts) {
+
+  // encode db name if opts.prefix is a url (#5574)
+  if (hasUrlPrefix(opts)) {
+    var dbName = opts.name.substr(opts.prefix.length);
+    name = opts.prefix + encodeURIComponent(dbName);
+  }
```

Comment:
I’ve tried it out but the tests pass in `opts.prefix` as `/path/to/pouchdb/tmp/_pouch_` which makes little sense to me as this is an http adapter?

Either way, the current solution works :)